Fix a potential panic issue

The following logs were reported when using the Agent deployed on a Windows Server node:

Looks like the agent panic, the error should be properly handed instead of panicking.
```
2018/10/04 23:03:37 [INFO] serf: EventMemberJoin: ee9b250aa26c 10.0.0.3,

2018/10/04 23:03:38 [WARN] memberlist: Failed to resolve tasks.agent: lookup tasks.agent: no such host,

2018/10/04 23:03:38 [INFO] - Starting Portainer agent version 1.1.2 on 0.0.0.0:9001 (cluster mode: true),

2018/10/04 23:04:05 http error: Missing Portainer public key (err=Portainer public key unavailable) (code=403),

2018/10/04 23:04:09 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

2018/10/04 23:04:09 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

2018/10/04 23:04:09 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

panic: interface conversion: interface {} is map[string]interface {}, not []interface {},

,

goroutine 90 [running]:,

bitbucket.org/portainer/agent/http/proxy.responseToJSONArray(0xc0422bce10, 0xc04214afe0, 0x9, 0x0, 0x0, 0x0, 0x0, 0x0),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/response.go:47 +0x35e,

bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).copyAndExecuteRequest(0xc042004138, 0xc04250f400, 0xc042031840, 0xc042207380, 0xc04214b010),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:101 +0x12a,

created by bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).executeRequestOnCluster,

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:79 +0x93,

2018/10/04 23:04:15 [INFO] serf: EventMemberJoin: ee9b250aa26c 10.0.0.4,

2018/10/04 23:04:16 [WARN] memberlist: Failed to resolve tasks.agent: lookup tasks.agent: no such host,

2018/10/04 23:04:16 [INFO] - Starting Portainer agent version 1.1.2 on 0.0.0.0:9001 (cluster mode: true),

2018/10/04 23:04:16 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

2018/10/04 23:04:16 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

panic: interface conversion: interface {} is map[string]interface {}, not []interface {},

,

goroutine 47 [running]:,

bitbucket.org/portainer/agent/http/proxy.responseToJSONArray(0xc0422bcc60, 0xc0420a9390, 0xc, 0x0, 0x0, 0x0, 0x0, 0x0),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/response.go:47 +0x35e,

bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).copyAndExecuteRequest(0xc042004138, 0xc04207f000, 0xc042031540, 0xc042206060, 0xc0420a93c0),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:101 +0x12a,

created by bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).executeRequestOnCluster,

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:79 +0x93,

2018/10/04 23:04:22 [INFO] serf: EventMemberJoin: ee9b250aa26c 10.0.0.5,

2018/10/04 23:04:23 [WARN] memberlist: Failed to resolve tasks.agent: lookup tasks.agent: no such host,

2018/10/04 23:04:23 [INFO] - Starting Portainer agent version 1.1.2 on 0.0.0.0:9001 (cluster mode: true),

2018/10/04 23:04:43 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

panic: interface conversion: interface {} is map[string]interface {}, not []interface {},

,

goroutine 53 [running]:,

bitbucket.org/portainer/agent/http/proxy.responseToJSONArray(0xc0422bd200, 0xc04214a070, 0xc, 0x0, 0x0, 0xc04212fef8, 0x42bc0d, 0x42e712),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/response.go:47 +0x35e,

bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).copyAndExecuteRequest(0xc042004138, 0xc04207f300, 0xc042030d40, 0xc0420330e0, 0xc04214a0a0),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:101 +0x12a,

created by bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).executeRequestOnCluster,

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:79 +0x93,

2018/10/04 23:04:48 [INFO] serf: EventMemberJoin: ee9b250aa26c 10.0.0.6,

2018/10/04 23:04:49 [WARN] memberlist: Failed to resolve tasks.agent: lookup tasks.agent: no such host,

2018/10/04 23:04:49 [INFO] - Starting Portainer agent version 1.1.2 on 0.0.0.0:9001 (cluster mode: true),

2018/10/04 23:04:51 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

2018/10/04 23:04:51 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

panic: interface conversion: interface {} is map[string]interface {}, not []interface {},

,

goroutine 34 [running]:,

bitbucket.org/portainer/agent/http/proxy.responseToJSONArray(0xc0422bcab0, 0xc0420a8b20, 0x10, 0x0, 0x0, 0x412672, 0xc04205ecb8, 0x8),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/response.go:47 +0x35e,

bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).copyAndExecuteRequest(0xc042004138, 0xc04207e700, 0xc042030480, 0xc042032120, 0xc0420a8bf0),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:101 +0x12a,

created by bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).executeRequestOnCluster,

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:79 +0x93,

2018/10/04 23:04:56 [INFO] serf: EventMemberJoin: ee9b250aa26c 10.0.0.7,

2018/10/04 23:04:57 [WARN] memberlist: Failed to resolve tasks.agent: lookup tasks.agent: no such host,

2018/10/04 23:04:57 [INFO] - Starting Portainer agent version 1.1.2 on 0.0.0.0:9001 (cluster mode: true),

2018/10/04 23:05:03 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

2018/10/04 23:05:03 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

panic: interface conversion: interface {} is map[string]interface {}, not []interface {},

,

goroutine 35 [running]:,

bitbucket.org/portainer/agent/http/proxy.responseToJSONArray(0xc0422bc7e0, 0xc04214a500, 0x10, 0x0, 0x0, 0x412672, 0xc04205cd88, 0x8),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/response.go:47 +0x35e,

bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).copyAndExecuteRequest(0xc042004138, 0xc04207e500, 0xc04227d2c0, 0xc042207f20, 0xc04214a570),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:101 +0x12a,

created by bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).executeRequestOnCluster,

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:79 +0x93,

2018/10/04 23:05:08 [INFO] serf: EventMemberJoin: ee9b250aa26c 10.0.0.8,

2018/10/04 23:05:09 [WARN] memberlist: Failed to resolve tasks.agent: lookup tasks.agent: no such host,

2018/10/04 23:05:09 [INFO] - Starting Portainer agent version 1.1.2 on 0.0.0.0:9001 (cluster mode: true),

2018/10/04 23:05:10 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

2018/10/04 23:05:10 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

2018/10/04 23:05:10 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

2018/10/04 23:05:10 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

panic: interface conversion: interface {} is map[string]interface {}, not []interface {},

,

goroutine 42 [running]:,

bitbucket.org/portainer/agent/http/proxy.responseToJSONArray(0xc0422bcea0, 0xc0420a97f0, 0x10, 0x0, 0x0, 0x0, 0x0, 0x0),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/response.go:47 +0x35e,

bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).copyAndExecuteRequest(0xc042004138, 0xc04207ef00, 0xc0420304c0, 0xc0420327e0, 0xc0420a9820),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:101 +0x12a,

created by bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).executeRequestOnCluster,

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:79 +0x93,

2018/10/04 23:05:15 [INFO] serf: EventMemberJoin: ee9b250aa26c 10.0.0.9,

2018/10/04 23:05:16 [WARN] memberlist: Failed to resolve tasks.agent: lookup tasks.agent: no such host,

2018/10/04 23:05:16 [INFO] - Starting Portainer agent version 1.1.2 on 0.0.0.0:9001 (cluster mode: true),

2018/10/04 23:05:17 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

2018/10/04 23:05:17 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

panic: interface conversion: interface {} is map[string]interface {}, not []interface {},

,

goroutine 78 [running]:,

bitbucket.org/portainer/agent/http/proxy.responseToJSONArray(0xc0422bc2d0, 0xc0421a87e0, 0x10, 0x0, 0x0, 0xc042233ef8, 0x42bc0d, 0x42e712),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/response.go:47 +0x35e,

bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).copyAndExecuteRequest(0xc042004138, 0xc042510e00, 0xc042031d00, 0xc0422072c0, 0xc0421a8810),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:101 +0x12a,

created by bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).executeRequestOnCluster,

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:79 +0x93,

2018/10/04 23:05:23 [INFO] serf: EventMemberJoin: ee9b250aa26c 10.0.0.10,

2018/10/04 23:05:24 [WARN] memberlist: Failed to resolve tasks.agent: lookup tasks.agent: no such host,

2018/10/04 23:05:24 [INFO] - Starting Portainer agent version 1.1.2 on 0.0.0.0:9001 (cluster mode: true),

2018/10/04 23:05:24 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

2018/10/04 23:05:24 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

2018/10/04 23:05:24 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

panic: interface conversion: interface {} is map[string]interface {}, not []interface {},

,

goroutine 33 [running]:,

bitbucket.org/portainer/agent/http/proxy.responseToJSONArray(0xc0422bcf30, 0xc04214b290, 0x10, 0x0, 0x0, 0x412672, 0xc04205ecb8, 0x8),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/response.go:47 +0x35e,

bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).copyAndExecuteRequest(0xc042004140, 0xc04207e400, 0xc042031040, 0xc042033a40, 0xc04214b300),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:101 +0x12a,

created by bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).executeRequestOnCluster,

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:79 +0x93,

2018/10/04 23:05:31 [INFO] serf: EventMemberJoin: ee9b250aa26c 10.0.0.11,

2018/10/04 23:05:32 [WARN] memberlist: Failed to resolve tasks.agent: lookup tasks.agent: no such host,

2018/10/04 23:05:32 [INFO] - Starting Portainer agent version 1.1.2 on 0.0.0.0:9001 (cluster mode: true),

2018/10/04 23:05:34 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

2018/10/04 23:05:34 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

2018/10/04 23:05:34 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

panic: interface conversion: interface {} is map[string]interface {}, not []interface {},

,

goroutine 71 [running]:,

bitbucket.org/portainer/agent/http/proxy.responseToJSONArray(0xc0422bd560, 0xc0421a8740, 0x10, 0x0, 0x0, 0x0, 0x0, 0x0),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/response.go:47 +0x35e,

bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).copyAndExecuteRequest(0xc042004138, 0xc042524a00, 0xc04228ab40, 0xc04253e720, 0xc0421a8770),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:101 +0x12a,

created by bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).executeRequestOnCluster,

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:79 +0x93,

2018/10/04 23:05:41 [INFO] serf: EventMemberJoin: ee9b250aa26c 10.0.0.12,

2018/10/04 23:05:42 [WARN] memberlist: Failed to resolve tasks.agent: lookup tasks.agent: no such host,

2018/10/04 23:05:42 [INFO] - Starting Portainer agent version 1.1.2 on 0.0.0.0:9001 (cluster mode: true),

2018/10/04 23:05:49 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

2018/10/04 23:05:49 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

2018/10/04 23:05:49 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

panic: interface conversion: interface {} is map[string]interface {}, not []interface {},

,

goroutine 42 [running]:,

bitbucket.org/portainer/agent/http/proxy.responseToJSONArray(0xc0422bcbd0, 0xc04214a340, 0x10, 0x0, 0x0, 0x0, 0x0, 0x0),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/response.go:47 +0x35e,

bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).copyAndExecuteRequest(0xc042004138, 0xc04207eb00, 0xc0422891c0, 0xc0420320c0, 0xc04214a370),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:101 +0x12a,

created by bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).executeRequestOnCluster,

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:79 +0x93,

2018/10/04 23:05:55 [INFO] serf: EventMemberJoin: ee9b250aa26c 10.0.0.13,

2018/10/04 23:05:56 [WARN] memberlist: Failed to resolve tasks.agent: lookup tasks.agent: no such host,

2018/10/04 23:05:56 [INFO] - Starting Portainer agent version 1.1.2 on 0.0.0.0:9001 (cluster mode: true),

2018/10/04 23:07:31 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

panic: interface conversion: interface {} is map[string]interface {}, not []interface {},

,

goroutine 91 [running]:,

bitbucket.org/portainer/agent/http/proxy.responseToJSONArray(0xc0422bdc20, 0xc0422a1a70, 0x10, 0x0, 0x0, 0x0, 0x0, 0x0),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/response.go:47 +0x35e,

bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).copyAndExecuteRequest(0xc042004140, 0xc0423aae00, 0xc042375680, 0xc0423dc4e0, 0xc0422a1b80),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:101 +0x12a,

created by bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).executeRequestOnCluster,

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:79 +0x93,

2018/10/04 23:07:36 [INFO] serf: EventMemberJoin: ee9b250aa26c 10.0.0.14,

2018/10/04 23:07:37 [WARN] memberlist: Failed to resolve tasks.agent: lookup tasks.agent: no such host,

2018/10/04 23:07:37 [INFO] - Starting Portainer agent version 1.1.2 on 0.0.0.0:9001 (cluster mode: true),

2018/10/04 23:07:38 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

2018/10/04 23:07:38 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

2018/10/04 23:07:38 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

panic: interface conversion: interface {} is map[string]interface {}, not []interface {},

,

goroutine 41 [running]:,

bitbucket.org/portainer/agent/http/proxy.responseToJSONArray(0xc0422bcd80, 0xc0420a96e0, 0x10, 0x0, 0x0, 0x0, 0x0, 0x0),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/response.go:47 +0x35e,

bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).copyAndExecuteRequest(0xc042004138, 0xc04207eb00, 0xc042030440, 0xc042032360, 0xc0420a9710),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:101 +0x12a,

created by bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).executeRequestOnCluster,

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:79 +0x93,

2018/10/04 23:07:43 [INFO] serf: EventMemberJoin: ee9b250aa26c 10.0.0.15,

2018/10/04 23:07:44 [WARN] memberlist: Failed to resolve tasks.agent: lookup tasks.agent: no such host,

2018/10/04 23:07:44 [INFO] - Starting Portainer agent version 1.1.2 on 0.0.0.0:9001 (cluster mode: true),

2018/10/04 23:08:51 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

2018/10/05 22:15:50 [INFO] serf: EventMemberJoin: ee9b250aa26c 10.0.0.17,

2018/10/05 22:15:51 [WARN] memberlist: Failed to resolve tasks.agent: lookup tasks.agent: no such host,

2018/10/05 22:15:51 [INFO] - Starting Portainer agent version 1.1.2 on 0.0.0.0:9001 (cluster mode: true),

2018/10/05 22:16:34 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

2018/10/05 22:16:34 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

panic: interface conversion: interface {} is map[string]interface {}, not []interface {},

,

goroutine 67 [running]:,

bitbucket.org/portainer/agent/http/proxy.responseToJSONArray(0xc0422bd3b0, 0xc04214b1a0, 0x10, 0x0, 0x0, 0xc04227a180, 0x50, 0xc042182138),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/response.go:47 +0x35e,

bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).copyAndExecuteRequest(0xc0420040a8, 0xc04223a500, 0xc04227c600, 0xc042207080, 0xc04214b1d0),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:101 +0x12a,

created by bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).executeRequestOnCluster,

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:79 +0x93,

2018/10/05 22:16:39 [INFO] serf: EventMemberJoin: ee9b250aa26c 10.0.0.18,

2018/10/05 22:16:40 [WARN] memberlist: Failed to resolve tasks.agent: lookup tasks.agent: no such host,

2018/10/05 22:16:40 [INFO] - Starting Portainer agent version 1.1.2 on 0.0.0.0:9001 (cluster mode: true),

2018/10/05 22:16:41 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

panic: interface conversion: interface {} is map[string]interface {}, not []interface {},

,

goroutine 44 [running]:,

bitbucket.org/portainer/agent/http/proxy.responseToJSONArray(0xc0422bcbd0, 0xc04214a7f0, 0x10, 0x0, 0x0, 0xc04236ec00, 0xc04236ebd0, 0xc04236e6f0),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/response.go:47 +0x35e,

bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).copyAndExecuteRequest(0xc042004138, 0xc04207ee00, 0xc042031340, 0xc042033f80, 0xc04214a840),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:101 +0x12a,

created by bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).executeRequestOnCluster,

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:79 +0x93,

2018/10/05 22:16:47 [INFO] serf: EventMemberJoin: ee9b250aa26c 10.0.0.19,

2018/10/05 22:16:48 [WARN] memberlist: Failed to resolve tasks.agent: lookup tasks.agent: no such host,

2018/10/05 22:16:48 [INFO] - Starting Portainer agent version 1.1.2 on 0.0.0.0:9001 (cluster mode: true),

2018/10/08 10:15:06 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

2018/10/08 10:15:06 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

2018/10/08 10:15:06 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

panic: interface conversion: interface {} is map[string]interface {}, not []interface {},

,

routine 12310 [running]:,

bitbucket.org/portainer/agent/http/proxy.responseToJSONArray(0xc0422bd4d0, 0xc0420182b0, 0x10, 0x0, 0x0, 0x0, 0x0, 0x0),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/response.go:47 +0x35e,

bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).copyAndExecuteRequest(0xc042004138, 0xc0423f1900, 0xc042031f00, 0xc042033aa0, 0xc0420182f0),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:101 +0x12a,

created by bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).executeRequestOnCluster,

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:79 +0x93,

2018/10/08 10:15:12 [INFO] serf: EventMemberJoin: ee9b250aa26c 10.0.0.20,

2018/10/08 10:15:13 [WARN] memberlist: Failed to resolve tasks.agent: lookup tasks.agent: no such host,

2018/10/08 10:15:13 [INFO] - Starting Portainer agent version 1.1.2 on 0.0.0.0:9001 (cluster mode: true),

2018/10/08 10:15:13 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

2018/10/08 10:15:13 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

panic: interface conversion: interface {} is map[string]interface {}, not []interface {},

,

goroutine 48 [running]:,

bitbucket.org/portainer/agent/http/proxy.responseToJSONArray(0xc0422bce10, 0xc04214b960, 0x10, 0x0, 0x0, 0xc04204def8, 0x42bc0d, 0x42e712),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/response.go:47 +0x35e,

bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).copyAndExecuteRequest(0xc042004140, 0xc04207f200, 0xc042031880, 0xc042206b40, 0xc04214b9d0),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:101 +0x12a,

created by bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).executeRequestOnCluster,

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:79 +0x93,

2018/10/08 10:15:18 [INFO] serf: EventMemberJoin: ee9b250aa26c 10.0.0.21,

2018/10/08 10:15:19 [WARN] memberlist: Failed to resolve tasks.agent: lookup tasks.agent: no such host,

2018/10/08 10:15:19 [INFO] - Starting Portainer agent version 1.1.2 on 0.0.0.0:9001 (cluster mode: true),

2018/10/08 10:15:33 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

2018/10/08 10:15:33 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

2018/10/08 10:15:41 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

panic: interface conversion: interface {} is map[string]interface {}, not []interface {},

,

goroutine 139 [running]:,

bitbucket.org/portainer/agent/http/proxy.responseToJSONArray(0xc042448bd0, 0xc0422a1780, 0x9, 0x0, 0x0, 0xc0423b3ef8, 0x42bc0d, 0x42e712),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/response.go:47 +0x35e,

bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).copyAndExecuteRequest(0xc042004138, 0xc042208b00, 0xc04222c200, 0xc0422777a0, 0xc0422a1a70),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:101 +0x12a,

created by bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).executeRequestOnCluster,

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:79 +0x93,

2018/10/08 10:15:47 [INFO] serf: EventMemberJoin: ee9b250aa26c 10.0.0.22,

2018/10/08 10:15:48 [WARN] memberlist: Failed to resolve tasks.agent: lookup tasks.agent: no such host,

2018/10/08 10:15:48 [INFO] - Starting Portainer agent version 1.1.2 on 0.0.0.0:9001 (cluster mode: true),

2018/10/08 10:15:48 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

2018/10/08 10:15:48 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

2018/10/08 10:15:48 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

panic: interface conversion: interface {} is map[string]interface {}, not []interface {},

,

goroutine 34 [running]:,

bitbucket.org/portainer/agent/http/proxy.responseToJSONArray(0xc0422bcbd0, 0xc04214a140, 0xc, 0x0, 0x0, 0x412672, 0xc04205ebe8, 0x8),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/response.go:47 +0x35e,

bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).copyAndExecuteRequest(0xc042004138, 0xc04207e700, 0xc042030480, 0xc042032120, 0xc04214a200),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:101 +0x12a,

created by bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).executeRequestOnCluster,

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:79 +0x93,

2018/10/08 10:15:54 [INFO] serf: EventMemberJoin: ee9b250aa26c 10.0.0.23,

2018/10/08 10:15:55 [WARN] memberlist: Failed to resolve tasks.agent: lookup tasks.agent: no such host,

2018/10/08 10:15:55 [INFO] - Starting Portainer agent version 1.1.2 on 0.0.0.0:9001 (cluster mode: true),

2018/10/08 10:15:55 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

2018/10/08 10:15:55 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

panic: interface conversion: interface {} is map[string]interface {}, not []interface {},

,

goroutine 34 [running]:,

bitbucket.org/portainer/agent/http/proxy.responseToJSONArray(0xc0422bcea0, 0xc04214a440, 0x10, 0x0, 0x0, 0x412672, 0xc04205ae58, 0x8),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/response.go:47 +0x35e,

bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).copyAndExecuteRequest(0xc042004138, 0xc04207e400, 0xc042375e00, 0xc0423af980, 0xc04214a4b0),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:101 +0x12a,

created by bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).executeRequestOnCluster,

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:79 +0x93,

2018/10/08 10:16:01 [INFO] serf: EventMemberJoin: ee9b250aa26c 10.0.0.24,

2018/10/08 10:16:02 [WARN] memberlist: Failed to resolve tasks.agent: lookup tasks.agent: no such host,

2018/10/08 10:16:02 [INFO] - Starting Portainer agent version 1.1.2 on 0.0.0.0:9001 (cluster mode: true),

2018/10/08 10:16:03 http error: Unable to proxy the request via the Docker socket (err=open //./pipe/docker_engine: The system cannot find the file specified.) (code=500),

panic: interface conversion: interface {} is map[string]interface {}, not []interface {},

,

goroutine 39 [running]:,

bitbucket.org/portainer/agent/http/proxy.responseToJSONArray(0xc0422bcd80, 0xc04214af40, 0x10, 0x0, 0x0, 0xf370a1c8ce872167, 0x13b69a62f24afc32, 0x91fa9cd51aad3c6c),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/response.go:47 +0x35e,

bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).copyAndExecuteRequest(0xc042004110, 0xc04207e900, 0xc042031300, 0xc0422060c0, 0xc04214af70),

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:101 +0x12a,

created by bitbucket.org/portainer/agent/http/proxy.(*ClusterProxy).executeRequestOnCluster,

                /home/tony/workspaces/go/src/bitbucket.org/portainer/agent/http/proxy/cluster.go:79 +0x93,

2018/10/08 10:16:09 [INFO] serf: EventMemberJoin: ee9b250aa26c 10.0.0.25,

2018/10/08 10:16:10 [WARN] memberlist: Failed to resolve tasks.agent: lookup tasks.agent: no such host,

2018/10/08 10:16:10 [INFO] - Starting Portainer agent version 1.1.2 on 0.0.0.0:9001 (cluster mode: true),
```

--- a/http/proxy/response.go
+++ b/http/proxy/response.go
@@ -3,6 +3,7 @@ package proxy
 import (
 	"encoding/json"
 	"errors"
+	"log"
 	"net/http"
 	"strings"
 )
@@ -44,7 +45,11 @@ func responseToJSONArray(response *http.Response, requestPath string) ([]interfa
 			responseData = make([]interface{}, 0)
 		}
 	} else {
-		responseData = responseObject.([]interface{})
+		responseData, ok = responseObject.([]interface{})
+		if !ok {
+			log.Printf("[ERROR] [http,proxy,cluster] [message: unexpected response from Docker daemon] [response: %+v]", responseData)
+			return nil, errors.New("invalid response from Docker daemon")
+		}
 	}
 
 	return responseData, nil

